src/lib389/lib389/replica.py

```diff
@@ -848,7 +848,10 @@ def __init__(self, ruvs=[], logger=None):
                 # Don't add rids if they have no csn (no writes) yet.
                 rid = pr[1]
                 self._rids.append(rid)
-                self._rid_url[rid] = pr[2]
+                try:
+                    self._rid_url[rid] = pr[2]
+                except IndexError:
+                    continue
```

Comment:
Don't "continue" here because we've already added the rid to Replica "_rids".  Instead set it to some default value, like what we do with other settings when they are missing.  Or, remove the rid from the "_rids" list before continuing.

Reply:
> Don't "continue" here because we've already added the rid to Replica "_rids". Instead set it to some default value, like what we do with other settings when they are missing. Or, remove the rid from the "_rids" list before continuing.

I would prefer to remove the rid as I'm afraid setting a default url might break something down the line.

Reply:
Agreed